File: 프로그래머스/1/12910. 나누어 떨어지는 숫자 배열/나누어 떨어지는 숫자 배열.cs
using System;
using System.Collections.Generic;
using System.Linq;

public class Solution 
{
    public int[] solution(int[] arr, int divisor) 
    {
        List<int> results = new List<int>();
        foreach(var n in arr)
        {
            if(n%divisor == 0)
            {
                results.Add(n);
            }
        }
        int[] ans = results.ToArray();
        Array.Sort(ans);
        return ans.Length == 0 ? new int[] {-1} : ans;
    }
}

File: 프로그래머스/1/12910. 나누어 떨어지는 숫자 배열/나누어 떨어지는 숫자 배열.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

vector<int> solution(vector<int> arr, int divisor) 
{
    vector<int> vec;
    for(int &i : arr)
    {
        if(i%divisor == 0)
        {
            vec.push_back(i);
        }
    }
    sort(vec.begin(), vec.end());
    if(vec.empty())
    {
        vec.push_back(-1);
    }
    return vec;
}